zmq.hpp

```diff
@@ -103,18 +103,28 @@
 #include <tuple>
 #include <memory>
 #endif
+
 #ifdef ZMQ_CPP17
 #ifdef __has_include
 #if __has_include(<optional>)
-#include <optional>
-#define ZMQ_HAS_OPTIONAL 1
+#define CPPZMQ_HAS_OPTIONAL 1
```

Comment:
If you want to make this overridable by a user, this needs to be guarded by an `#ifndef CPPZMQ_HAS_OPTIONAL`.